src/lib.rs

```diff
@@ -213,11 +214,13 @@ pub mod pallet {
 			Supersigs::<T>::insert(index, supersig);
 			NonceSupersig::<T>::put(index + 1);
 
-			for _ in 0..members_len {
-				//Err if the account's provider is 0, but we just created an account, and
-				//transfered the existencial deposit, so provider will be 1
-				let _res = frame_system::Pallet::<T>::inc_consumers(&supersig_id);
+			// This mean the supersig account cannot be emptied while existing in this storage
+			frame_system::Pallet::<T>::inc_consumers(&supersig_id)?;
+			// This mean that the members account cannot be emptied while there are member of a supersig
+			for member in members {
+				frame_system::Pallet::<T>::inc_consumers(&member)?;
 			}
```

Comment:
This is the key part.
Now apply this logic each time we add and remove and user from a multisig, and when we remove a multisig from storage